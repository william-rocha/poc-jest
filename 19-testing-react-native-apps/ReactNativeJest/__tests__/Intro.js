import React from "react";
import Intro from "../Intro";

jest.mock("react-native-video", () => "Video");

jest.mock("Text", () => {
  const RealComponent = jest.requireActual("Text");
  const React = require("React");
  class Text extends React.Component {
    render() {
      return React.createElement("Text", this.props, <div>Mock</div>);
    }
  }
  Text.propTypes = RealComponent.propTypes;
  return Text;
});

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});
