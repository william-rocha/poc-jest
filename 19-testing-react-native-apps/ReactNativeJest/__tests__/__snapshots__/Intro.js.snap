// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Video
    source={
      Object {
        "uri": "background",
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    <div>
      Mock
    </div>
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    <div>
      Mock
    </div>
  </Text>
</View>
`;
